pkg/updater/resultstore/resultstore.go

```diff
@@ -70,6 +73,117 @@ type processedResult struct {
 	TargetResults   map[string][]*singleActionResult
 }
 
+type timedError struct {
+	millis int64
+	err    error
+}
+type invocationGroup struct {
+	group       string
+	invocations []processedResult
+	errs        map[string]timedError
+}
+
+const (
+	// PartOf is the configuration value key which links to the group invocation ID.
+	PartOf = "EXECUTED_AS_PART_OF_INVOCATION"
+	// Originally indicates the associated non-overview indication.
+	Originally = "Original_test_invocation"
+)
+
+// buildIdentifier may assume its arguments are non-nil.
+type buildIdentifier func(configpb.TestGroup, processedResult) string
+
+var (
+	zero = time.Unix(0, 0)
+)
+
+func timeRunDate(milliseconds int64) time.Time {
+	return zero.Add(time.Duration(milliseconds) * time.Millisecond)
+}
+
+func invocationGroupID(inv processedResult, primary configpb.TestGroup_PrimaryGrouping, fallback configpb.TestGroup_FallbackGrouping, fallbackKey string, identifyBuild string) string {
+	if primary == configpb.TestGroup_PRIMARY_GROUPING_BUILD {
+		return identifyBuild
+	}
+	// PRIMARY_GROUPING_NONE, consider fallback
+	id := inv.InvocationProto.GetId().String()
+	for _, cfg := range inv.InvocationProto.GetProperties() {
+		// TODO(amerai): Group by workflow ID instead.
+		if cfg.GetKey() != PartOf {
+			continue
+		}
+		parts := strings.Split(cfg.GetValue(), "/")
+		if groupID := parts[len(parts)-1]; groupID != id {
+			return groupID
+		}
+		break
+	}
+	switch fallback {
+	case configpb.TestGroup_FALLBACK_GROUPING_DATE:
+		return timeRunDate(inv.InvocationProto.GetTiming().StartTime.GetSeconds()).UTC().Format("gid-2006-01-02")
+	case configpb.TestGroup_FALLBACK_GROUPING_LABELS:
+		h := fnv.New32a()
+		for _, l := range inv.InvocationProto.InvocationAttributes.GetLabels() {
+			h.Write([]byte(l))
+		}
+		return "gid-" + strconv.Itoa(int(h.Sum32()))
+	case configpb.TestGroup_FALLBACK_GROUPING_ID:
+		return id
+	case configpb.TestGroup_FALLBACK_GROUPING_BUILD:
+		return "gid-" // FIX THIS+ inv.cl // true even if overriding changelist
```

Comment:
you can use identify build function here

Reply:
ok, Sounds good